processing/src/main/java/org/apache/druid/query/select/SelectQueryQueryToolChest.java

```diff
@@ -289,22 +287,22 @@ public Object apply(final Result<SelectResultValue> input)
             DateTime timestamp = granularity.toDateTime(((Number) resultIter.next()).longValue());
 
             Map<String, Integer> pageIdentifier = jsonMapper.convertValue(
-                resultIter.next(), new TypeReference<Map<String, Integer>>() {}
-                );
-            Set<String> dimensionSet = jsonMapper.convertValue(
-                resultIter.next(), new TypeReference<Set<String>>() {}
-            );
-            Set<String> metricSet = jsonMapper.convertValue(
-                resultIter.next(), new TypeReference<Set<String>>() {}
+                resultIter.next(),
+                new TypeReference<Map<String, Integer>>() {}
```

Comment:
can these type references just be `static final` fields?

Reply:
I also noticed this thing recently. There are more than a hundred occasions in the codebase. Should be fixed in a separate PR, if anybody will have willpower to do that. Could be spotted with a simple regex `(?<!= )new TypeReference`

Reply:
#6772